src/main/java/com/cellery/api/backend/ui/controllers/ProductController.java

```diff
@@ -0,0 +1,111 @@
+package com.cellery.api.backend.ui.controllers;
+
+import com.cellery.api.backend.shared.ProductDto;
+import com.cellery.api.backend.shared.Util.MapperUtil;
+import com.cellery.api.backend.ui.model.request.CreateProductRequestModel;
+import com.cellery.api.backend.ui.model.request.DeleteProductRequestModel;
+import com.cellery.api.backend.ui.model.request.EditProductRequestModel;
+import com.cellery.api.backend.ui.model.request.ProductRequestModel;
+import com.cellery.api.backend.ui.model.response.ProductRespModel;
+import com.cellery.api.backend.ui.service.ProductsService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+
+    private ProductsService productsService;
+    private MapperUtil modelMapper;
+
+    @Autowired
+    ProductController(ProductsService ps, MapperUtil mapper) {
+        this.productsService = ps;
+        this.modelMapper = mapper;
+    }
+
+    @GetMapping
+    public String test() {
+        return "You have poked the /products endpoint";
+    }
+
+    @GetMapping(path="/get")
+    public ResponseEntity<ProductRespModel> getProduct(@Valid @RequestBody ProductRequestModel productReq) {
```

Comment:
For the mappings, is it possible to have api routes such as
POST /products/users 
GET /products/users 
GET /products/users/{id}.
@GetMapping and such should be able to differentiate requests coming in even if they share the same route